### ISSUE TITLE
Expected n_neighbors <= n_samples,  but n_samples = 3, n_neighbors = 5

### ISSUE THREAD

**Issue by AdityaKrVerma** [ArthurLu/MLND.Project5#1]
when i run the knn model for 5000 tuples of the dataset....i get the above error. can u plz help me in solving this error.
the error occurs in the following code.

In [43]:

from sklearn.neighbors import KNeighborsClassifier

# Given cell_id, this function will train a KNN classifier on that cell and it returns the predicted as well as test labels. 
# Deuring training, it would not consider those place if the number of occurances is lower than the threshold.
def process_one_cell_KNN(df_train, cell_id, x_offset, y_offset, 
                         cols=None, fw=None, params={}):
    # Split data by time
    X_train, X_valid, y_train, y_valid = data_split(df_train, cell_id, x_offset, y_offset)
    
    # Feature Engineering
    if fw:
        X_train = feature_engineering_KNN(X_train, fw)
        X_valid = feature_engineering_KNN(X_valid, fw)

    # Converting into np array
    if cols:
        X_train = X_train[cols].values
        X_valid = X_valid[cols].values
    else:
        X_train = X_train.values
        X_valid = X_valid.values
    y_train = y_train.values
    y_valid = y_valid.values  

    # Creating KNN classifier
    clf = KNeighborsClassifier(**params)
    clf.fit(X_train, y_train)
    y_prob = pd.DataFrame(clf.predict_proba(X_valid), columns=clf.classes_)
    y_pred3 = [] # Predict top 3 place_ids
    y_pred1 = [] # Predict top 1 place_ids
    for _, row in y_prob.iterrows():
        data = row.sort_values(ascending=False).index[:3]
        y_pred3.append(data)
        y_pred1.append(data[:1])

    return y_pred3, np.array(y_pred1).reshape(1,-1)[0], y_valid

In [44]:

y_preds = []
y_valids = []
# Randomly select cells for validating
for cell_id in tqdm_notebook(cell_ids, total=n_validate_cells):
    _, y_pred, y_valid = process_one_cell_KNN(df_train, cell_id, x_offset, y_offset)
    y_preds.append(y_pred)
    y_valids.append(y_valid)
print "Accuracy of KNN in {} cells: {:.4f}".format(n_validate_cells, calculate_accuracy(y_preds, y_valids))

*i cant take all of the dataset due to limited resources. if taking a part of dataset is the issue...then minimum how much part should i consider for the code to work.
